src/ORM/Rule/IsUnique.php

```diff
@@ -86,23 +77,15 @@ public function __invoke(EntityInterface $entity, array $options): bool
     /**
      * Add a model alias to all the keys in a set of conditions.
      *
-     * Null values will be omitted from the generated conditions,
-     * as SQL UNIQUE indexes treat `NULL != NULL`
-     *
      * @param string $alias The alias to add.
      * @param array $conditions The conditions to alias.
-     * @param bool $multipleNulls Whether or not to allow multiple nulls.
      * @return array
      */
-    protected function _alias(string $alias, array $conditions, bool $multipleNulls): array
+    protected function _alias(string $alias, array $conditions): array
     {
         $aliased = [];
         foreach ($conditions as $key => $value) {
-            if ($multipleNulls) {
-                $aliased["$alias.$key"] = $value;
-            } else {
-                $aliased["$alias.$key IS"] = $value;
-            }
```

Comment:
I think this option was trying to enable the creation of unique rules that disallowed multiple null values in a multi-column index. While valid in SQL they are sometimes not what people wanted. With that said, if there is no test for it, or a broken test there isn't much we can do.

Reply:
Keeping an obscure feature without test coverage doesn't make sense, hence I removed it. If you prefer we can retain the `$options` argument for constructor so that we can re-add the option later if someone comes up with a proper test case.

Reply:
We might want to look into this also for 3.x then as this seems to have similar issues then there.

Reply:
Refs
- https://github.com/cakephp/cakephp/pull/11029/files
- https://github.com/cakephp/cakephp/pull/8621/files (!)
- https://github.com/cakephp/docs/pull/4954/files (!)

Reply:
#11029 if for `Existsin`. The `allowNullableNulls` option there does seem to have proper test coverage.

https://github.com/cakephp/cakephp/pull/8621#issuecomment-217716001 probably sheds light on the fact that `allowMultipleNulls` option was introduced to fix generation of invalid `foo = null`. It isn't actually a new feature. Now since the `= null` problem is solved by simply using `'foo IS'` for all fields, there's no need for that option.

Reply:
https://stackoverflow.com/questions/36031664/cakephp-3-multiple-isunique-allowing-null-duplicate